### PR TITLE
Use byAgentURL for wmstats updates; create smaller chunks to update - wmagent 1.1.15 version

### PR DESCRIPTION
**Pull request by amaltaro**
Superseeds #8873

wmagent branch version of #8907 on top of 8873, but with the try/except block added to make sure it doesn't crash if it finds not well formatted docs (honestly, there should be none).

### CODE REVIEW

**Comment by amaltaro** [comment]
For the record, it took less than 7 secs to upload 98 docs to wmstats testbed (vocms0192):
```
2018-11-07 11:00:20,972:140364380997376:INFO:WMStatsWriter:Committing bulk of 98 docs ...
2018-11-07 11:00:27,165:140364380997376:INFO:AnalyticsPoller:Request data upload success
```

maybe getting all documents well formatted and dumping the wmstats production database to have a fresh start will help.

**Comment by amaltaro** [comment]
[Fix #8909 - Use byAgentURL for wmstats updates; create smaller chunks to update](https://trello.com/c/qocsDy3W/133-fix-8909-use-byagenturl-for-wmstats-updates-create-smaller-chunks-to-update)

File: src/python/WMCore/Services/WMStats/WMStatsWriter.py [hunk 1]
```diff
@@ -80,68 +82,30 @@ def uploadData(self, docs):
...
-                        else:
-                            raise Exception("wrong id %s" % doc['_id'])
-
-                    notInCacheKey.append([agentURL, idParts[1]])
-                    notInCacheDoc.append(doc)
-
-        self.couchDB.commit(new_edits=False)
-
+                    # just send well formatted revision for the new documents which required by new_edits=False
+                    doc['_revisions'] = {"start": 1, "ids": ["1234567890"]}
```
Comment (ticoann) [inline src/python/WMCore/Services/WMStats/WMStatsWriter.py]: Actually, this doesn't seems to work.  it has to be doc['_rev'] for initial setting.

**Comment by amaltaro** [comment]
Seangchan, as we discussed yesterday, this patch seems to be working fine. I'm squashing the last commit and will start pushing it to production. Then will ask Lina to dump the wmstats database which is now 331 GB !!!
I'm updating the description in the first comment too.

**Comment by amaltaro** [comment]
All the agents connected to cmsweb production have been patched.

**Comment by ticoann** [comment]
Thanks Alan  I am merging this patch.

**Pull request merged**